cocotb/triggers.py

```diff
@@ -456,13 +456,20 @@ def __repr__(self):
 class Lock(object):
     """Lock primitive (not re-entrant).
 
-    This should be used as::
+    This can be used as::
 
-        yield lock.acquire()
+        await lock.acquire()
         try:
             # do some stuff
         finally:
             lock.release()
+
+    The lock can be used as an asynchronous
+    context manager in a :keyword:`with`,
+    statement::
+
+        async with lock:
+            # do some stuff
```

Comment:
Ir's important to get the `versionchanged` here and a mention of python 3.6

```suggestion
    .. versionchanged:: 1.4
        The lock can be used as an asynchronous context manager in a
        :keyword:`with`, statement, provided Python 3.6 or newer is used::

            async with lock:
                # do some stuff
```
Also rewrapped the lines, since this is unusually narrow.

Reply:
Okay. I should point out asynchronous context management was introduced in Python3.5 https://docs.python.org/3/whatsnew/3.5.html.

Reply:
Ah, that's a very good point. Then no need to mention the python version.

Still need `versionchanged` though.